src/Components/Web.JS/src/Platform/Mono/MonoPlatform.ts

```diff
@@ -249,16 +249,39 @@ async function createEmscriptenModuleInstance(resourceLoader: WebAssemblyResourc
   const existingPostRun = moduleConfig.postRun || [];
   (moduleConfig as any).preloadPlugins = [];
 
+  var totalResources = Object.keys(resources.assembly).length + Object.keys(resources.pdb || {}).length + /* dotnet.wasm */ 1;
+
+  if (resourceLoader.bootConfig.resources.runtime.hasOwnProperty('dotnet.timezones.blat')){
+    totalResources++;
+  }
+
+  if (resourceLoader.bootConfig.icuDataMode !== ICUDataMode.Invariant){
+    totalResources++;
+  }
+
+  var resourcesLoaded = 0;
+  function setProgress(){
+    resourcesLoaded++;
+    const targetElement = document.getElementById('app') as HTMLDivElement;
+    const percentage = resourcesLoaded / totalResources * 100;
+    targetElement.style.setProperty('--blazor-load-percentage', percentage + '%');
+    document.getElementById('progress')!.style.display = 'block';
+    document.getElementById('percentage')!.textContent = Math.floor(percentage) + '%';
+  }
+
   // Begin loading the .dll/.pdb/.wasm files, but don't block here. Let other loading processes run in parallel.
   const dotnetWasmResourceName = 'dotnet.wasm';
   const assembliesBeingLoaded = resourceLoader.loadResources(resources.assembly, filename => `_framework/${filename}`, 'assembly');
+  assembliesBeingLoaded.forEach(loadingResource => loadingResource.response.then(_ => setProgress()));
```

Comment:
Would the following work?

```suggestion
  assembliesBeingLoaded.forEach(loadingResource => loadingResource.response.then(setProgress));
```

Reply:
When I made this change, the setProgress function was never actually reached. I can keep it as it currently is.